datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/package.scala

```diff
@@ -42,8 +42,42 @@ package object plans {
       expr.references.subsetOf(plan.outputSet)
     }
 
+
+    /**
+     * If exp is a ScalaUDF, then for each of it's children, we have to check if
+     * children can be derived from another scala UDF children from exprList
+     * @param exp scalaUDF
+     * @param exprList predicate and rejoin output list
+     * @return if udf can be derived from another udf
+     */
+    def canEvaluate(exp: ScalaUDF, exprList: Seq[Expression]): Boolean = {
+      val canBeDerived = false
+      exprList.forall {
+        case udf: ScalaUDF =>
+          if (udf.children.length == exp.children.length) {
+            var canUDFBeDerived = true
+            for (i <- udf.children.indices) {
```

Comment:
instead of setting first to true and then check in for loop to set false, can't we check children.forall, () and then just set to true if condition matches?  else anyway it will be false

Reply:
Changed to list1.zip(list2)